### DISCUSSION TITLE
Look forward to your support for Perplexity.ai.

### DISCUSSION THREAD

**Discussion by frankcsliu** [logancyang/obsidian-copilot#252] [Ideas]
It will be great to see if I can use the API key of Perplexity AI  in your add-on. Thank you!

**Comment by senseiBas** [comment]
Is there some way we can help make this happen?
Seems like an epic integration

**Comment by logancyang** [comment]
With this upcoming release you should be able to add any model that has the same OpenAI format by setting the name, URL and API key. #524

**Comment by GlebYavorski** [comment]
@logancyang thank you for letting us choose the base URL for OpenAI compatible APIs. Please, be aware, that Perplexity also provides us with special json tag 'citations' with the links to used sources. It is the core feature of using such online search models. Could you please somehow include this tag to the Copilot reply? Or just let us define some custom code to fetch it from the API reply?

```
$ curl -X POST --url https://api.perplexity.ai/chat/completions --header 'accept: application/json' --header 'content-type: application/json' --header "Authorization: Bearer ${PERPLEXITY_API_KEY}" --data '{
  "model": "llama-3.1-sonar-huge-128k-online",
  "messages": [
    {
      "role": "user",
      "content": "Some question"
    }
  ]
}'
{"id": "ddc9f9d7-c424-4710-9cfd-fcaf924b6c6f", "model": "llama-3.1-sonar-huge-128k-online", "created": 1736150987, "usage": {"prompt_tokens": 22, "completion_tokens": 65, "total_tokens": 87}, "citations": ["https://zachestnyibiznes.kz/company/ul/1097746034022_7724698537_OOO-ESBILDER", "https://check.tochka.com/company/1097746034022/"], "object": "chat.completion", "choices": [{"index": 0, "finish_reason": "stop", "message": {"role": "assistant", "content": "Some reply [1][2][4]."}, "delta": {"role": "assistant", "content": ""}}]}```

**Reply by jhmonroe** [reply]
This would be great—curently they just show up as unlinked numbers after a sentence like this. [5]

One of the main reasons to use Perplexity... and aren't other models starting to include citations too

**Reply by Joolyan** [reply]
This feature exists in the free Raycast.app extension "Perplexity API" (https://www.raycast.com/ilian/perplexity-api) so hopefully should be relatively easy to implement in Copilot :)

The code for the extension is available at https://github.com/raycast/extensions/tree/95b2d45874ef0bcbd5c2bb7af1f30e786a7bc8fb/extensions/perplexity-api/
